_rules/user-motion-user-interface-9d42d6.md

```diff
@@ -0,0 +1,592 @@
+    <title>Failed Example 2</title>
+    <style>
+      div:first-child {
+        font-size: 1.5em;
+        text-align: center;
+        vertical-align: middle;
+        display: table-cell;
+        height: 50vh;
+        width: 100vw;
+      }
+      #target {
+        background: white;
+        border: 1px solid black;
+      }
+    </style>
+    <script>
+      let eventCache = new Array();
+      let prevDiff = -1;
+
+      function init() {
+        const target = document.getElementById('target');
+        target.onpointerdown = pointerdown_handler;
+        target.onpointermove = pointermove_handler;
+
+        target.onpointerup = pointerup_handler;
+        target.onpointercancel = pointerup_handler;
+        target.onpointerout = pointerup_handler;
+        target.onpointerleave = pointerup_handler;
+      }
+
+      function pointerdown_handler(event) {
+        eventCache.push(event);
+      }
+
+      function pointermove_handler(event) {
+        event.target.style.border = 'dashed';
+        for (let i = 0; i < eventCache.length; i++) {
+          if (event.pointerId === eventCache[i].pointerId) {
+            eventCache[i] = event;
+            break;
+          }
+        }
+
+        if (eventCache.length === 2) {
+          if (prevDiff > 0) {
+            const curDiff = Math.abs(eventCache[0].clientX - eventCache[1].clientX);
+
+            if (curDiff > prevDiff) {
+              event.target.style.background = 'pink';
+            } else if (curDiff < prevDiff) {
+              event.target.style.background = 'lightblue';
+            }
+          }
+
+          prevDiff = curDiff;
+        }
+      }
+
+      function pointerup_handler(event) {
+        remove_event(event);
+        event.target.style.background = 'white';
+        event.target.style.border = '1px solid black';
+
+        if (eventCache.length < 2) {
+          prevDiff = -1;
+        }
+      }
+
+      function remove_event(event) {
+        for (let i = 0; i < eventCache.length; i++) {
+          if (eventCache[i].pointerId === event.pointerId) {
+            eventCache.splice(i, 1);
+            break;
+          }
+        }
+      }
+
+      function changeBackgroundColor(color) {
+        const target = document.getElementById('target');
+        target.style.background = color;
+      }
+    </script>
+  </head>
+  <body onload='init();' style='touch-action:none'>
+    <div id='target'>Touch and Hold with 2 pointers, then pinch in or out horizontally.<br/>
+        The background color will change to pink if the pinch is opening (Zoom In) 
+        or changes to lightblue if the pinch is closing (Zoom out).
+    </div>
+    <div aria-hidden="true">
+      <button onclick="changeBackgroundColor('lightblue');">Change background color to lightblue</button>
+      <button onclick="changeBackgroundColor('pink');">Change background color to pink</button>
+      <button onclick="changeBackgroundColor('white');">Reset color</button>
+    </div>  
```

Comment:
What would happen if only one of the buttons were not accessible (or not visible, or empty name)? Would that fail the rule (I think it should)? It's probably worth updating the examples so that different parts of the UI are "out" in each one. Expectation can also be made more explicit about that ("**each** UI component controlling this must … blah blah").

Same goes for "device motion has UI control" rule.